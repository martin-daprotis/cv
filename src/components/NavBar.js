import React,{useState,useLayoutEffect} from "react";
import Toolbar from "@material-ui/core/Toolbar";
import AppBar from "@material-ui/core/AppBar";
import {
  makeStyles,
  Icon,
  Tab,
  Tabs,
  Box,
  IconButton,
  Avatar,
  Grid,
} from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  container: {
    margin: "auto",
    maxWidth: "70%",
  },
  topic: {
    margin: "auto",
    paddingTop: "100px",
  },
  small: {
    width: theme.spacing(2.5),
    height: theme.spacing(2.5),
  },
}));

export default function NavBar({
  t,
  onThemeTypeSwitch,
  handleClickLang,
  handleClickTab,
  tabValue,
}) {
  const classes = useStyles();
  const [themeType, setThemeType] = React.useState(true);
  const [icon, setIcon] = React.useState("brightness_2");
  const [scrollTop, setScrollTop] = useState(window.pageYOffset);

  const handleClick = (e) => {
    setThemeType(!themeType);
    onThemeTypeSwitch();
    setIcon(icon === "brightness_7" ? "brightness_2" : "brightness_7");
  };

  
  useLayoutEffect(() => {
    const onScroll = () =>  {
      let currentPosition = window.pageYOffset; // or use document.documentElement.scrollTop;
      setScrollTop(currentPosition <= 0 ? 0 : currentPosition);
    }
    window.addEventListener("scroll", onScroll);
    return window.addEventListener("scroll", onScroll);
  }, [scrollTop]);

  return (
    <div>
      <AppBar style={{backgroundColor: `rgba(0,151,167,${(scrollTop-450)/100})`}}>
        <Toolbar varian="condense" className={classes.titlebar}>
          <Grid container direction="row" justify="space-between">
            <Grid item xs={9}>
              <Box display="flex" justifyContent="flex-start">
                <Tabs
                  value={tabValue}
                  indicatorColor="secondary"
                  onChange={handleClickTab}
                  variant="scrollable"
                  scrollButtons="auto"
                  aria-label="scrollable auto tabs example"
                >
                  <Tab label={t("navbar.profile")} />
                  <Tab label={t("navbar.knowledge")} />
                  <Tab label={t("navbar.personal_skills")} />
                  <Tab label={t("navbar.work")} />
                </Tabs>
              </Box>
            </Grid>
            <Grid item xs={3}>
              <Box display="flex" justifyContent="flex-end">
                <IconButton onClick={handleClickLang}>
                  <Avatar className={classes.small}>
                    <Box fontWeight="fontWeightBold" fontSize={12} m={1}>
                      {t("lang") === "SP" ? "EN" : "SP"}
                    </Box>
                  </Avatar>
                </IconButton>
                <IconButton>
                  <Icon style={{color:"#FFF"}} onClick={handleClick}>{icon}</Icon>
                </IconButton>
              </Box>
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
    </div>
  );
}
