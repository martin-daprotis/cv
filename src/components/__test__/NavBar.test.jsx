import React from 'react'
import { render, fireEvent, cleanup, waitFor, getByTestId ,logDOM} from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import {
  translate,
  setTranslations,setDefaultLanguage
} from "react-switch-lang";
import NavBar from '../NavBar';


const mockEN = {
  "navbar": {
    "profile": "Profile",
    "personal_skills": "Personal Skills",
    "knowledge": "Knowledge",
    "work": "Work Experience"
  }
};
const mockSP = {
  "navbar": {
    "profile": "Perfil",
    "personal_skills": "Habilidades",
    "knowledge": "Conocimientos",
    "work": "Experiencia Laboral"
  }
};

afterEach(cleanup)

const toggleTheme = jest.fn();
const handleClickLang = jest.fn();
const handleClickTab = jest.fn();

describe('Spanish language specifying mocks', () => {
  it('render properly ',  () => {
    setTranslations({ mockEN, mockSP });
    setDefaultLanguage("mockSP");
    const RenderHOC = translate(({t}) => <NavBar
                                              onThemeTypeSwitch={toggleTheme}
                                              handleClickLang={handleClickLang}
                                              t={t}
                                              handleClickTab={handleClickTab}
                                              tabValue={0}
                                            />);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText(/perfil/i)).toBeInTheDocument();
    expect(getByText(/habilidades/i)).toBeInTheDocument();
    expect(getByText(/conocimientos/i)).toBeInTheDocument();
    expect(getByText(/experiencia laboral/i)).toBeInTheDocument();
  })
})

describe('English language specifying mocks', () => {
  it('render properly ',  () => {
    setTranslations({ mockEN, mockSP });
    setDefaultLanguage("mockEN");
    const RenderHOC = translate(({t}) => <NavBar
                                              onThemeTypeSwitch={toggleTheme}
                                              handleClickLang={handleClickLang}
                                              t={t}
                                              handleClickTab={handleClickTab}
                                              tabValue={0}
                                            />);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText(/profile/i)).toBeInTheDocument();
    expect(getByText(/personal skills/i)).toBeInTheDocument();
    expect(getByText(/knowledge/i)).toBeInTheDocument();
    expect(getByText(/work experience/i)).toBeInTheDocument();
  })
})

describe('Action tests', () => {
  it('handles click properly ',  () => {
    const RenderHOC = translate(({t}) => <NavBar
                                                onThemeTypeSwitch={toggleTheme}
                                                handleClickLang={handleClickLang}
                                                t={t}
                                                handleClickTab={handleClickTab}
                                                tabValue={0}
                                              />);
  const {getByTestId} = render(<RenderHOC/>)
  fireEvent.click(getByTestId('btnLang'))
  expect(handleClickLang.mock.calls.length).toBe(1)
  fireEvent.click(getByTestId('btnTheme'))
  expect(toggleTheme.mock.calls.length).toBe(1)

  // logDOM()
  // fireEvent.scroll(window, { target: { scrollY: 1500 } })
  // expect((await findAllByText('80%')).length).toBeTruthy();
  })
})

it('matches snapshot',  () => {
  const RenderHOC = translate(({t}) => <NavBar
                                              onThemeTypeSwitch={toggleTheme}
                                              handleClickLang={handleClickLang}
                                              t={t}
                                              handleClickTab={handleClickTab}
                                              tabValue={0}
                                            />);
const wrapper = render(<RenderHOC/>)
expect(wrapper.asFragment()).toMatchSnapshot()
})

