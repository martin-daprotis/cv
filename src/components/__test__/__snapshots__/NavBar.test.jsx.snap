// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      style="background-color: rgba(0, 151, 167, 0);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        varian="condense"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div
              class="MuiBox-root MuiBox-root-38"
            >
              <div
                class="MuiTabs-root"
                data-testid="navbarTab"
              >
                <div
                  class="MuiTabs-scrollable"
                  style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                />
                <div
                  class="MuiTabs-scroller MuiTabs-scrollable"
                  style="margin-bottom: 0px;"
                >
                  <div
                    aria-label="scrollable auto tabs example"
                    class="MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        Profile
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        Knowledge
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        Personal Skills
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        Work Experience
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="PrivateTabIndicator-root-39 PrivateTabIndicator-colorSecondary-41 MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              class="MuiBox-root MuiBox-root-43"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="btnLang"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-small-37 MuiAvatar-colorDefault"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-44"
                    >
                      SP
                    </div>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="btnTheme"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <span
                    aria-hidden="true"
                    class="material-icons MuiIcon-root"
                    style="color: rgb(255, 255, 255);"
                  >
                    brightness_2
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
